openapi/spec.go

```diff
@@ -184,6 +229,35 @@ type Schema struct {
 	Enum             []interface{} `json:"enum,omitempty" yaml:"enum,omitempty"`
 	Nullable         bool          `json:"nullable,omitempty" yaml:"nullable,omitempty"`
 	Deprecated       bool          `json:"deprecated,omitempty" yaml:"deprecated,omitempty"`
+
+	Extensions map[string]interface{} `json:"-" yaml:"-"`
+}
+
+func (s *Schema) MarshalJSON() ([]byte, error) {
```

Comment:
If I get it right, you're trying to simulate the new `omitnil` JSON field tag, to avoid marshaling the `Extensions` field it it's empty ? What the purpose.
`omitnil` could be used (from go1.24), and for previous versions, it doesn't really matter if we marshal `extensions: []`.